everest-testing/src/everest/testing/core_utils/configuration/libocpp_configuration_helper.py

```diff
@@ -14,62 +36,54 @@ def generate_ocpp_config(self,
                              target_ocpp_user_config_file: Path,
                              source_ocpp_config_file: Path,
                              central_system_host: str,
-                             central_system_port: Union[str, int]):
-        config = self._get_occp_config(central_system_port=central_system_port,
-                                       central_system_host=central_system_host,
-                                       source_ocpp_config_file=source_ocpp_config_file)
+                             central_system_port: Union[str, int],
+                             configuration_visitors: list[OCPPConfigAdjustmentVisitor] | None = None):
+        config = json.loads(source_ocpp_config_file.read_text())
+
+        configuration_visitors = configuration_visitors if configuration_visitors else []
+
+        for v in [self._get_default_visitor(central_system_port, central_system_host)] + configuration_visitors:
+            config = v.adjust_ocpp_configuration(config)
+
         with target_ocpp_config_file.open("w") as f:
             json.dump(config, f)
         target_ocpp_user_config_file.write_text("{}")
 
     @abstractmethod
-    def _get_occp_config(self, central_system_port, central_system_host, source_ocpp_config_file: Path):
+    def _get_default_visitor(self, central_system_port: int | str,
+                             central_system_host: str) -> OCPPConfigAdjustmentVisitor:
         pass
 
 
 class LibOCPP16ConfigurationHelper(LibOCPPConfigurationHelperBase):
+    def _get_default_visitor(self, central_system_port, central_system_host):
+        def adjust_ocpp_configuration(config: dict) -> dict:
+            config = deepcopy(config)
+            charge_point_id = config["Internal"]["ChargePointId"]
+            config["Internal"][
+                "CentralSystemURI"] = f"{central_system_host}:{central_system_port}/{charge_point_id}"
+            return config
 
-    def _get_occp_config(self,  central_system_port, central_system_host, source_ocpp_config_file: Path):
-        ocpp_config = json.loads(source_ocpp_config_file.read_text())
-        charge_point_id = ocpp_config["Internal"]["ChargePointId"]
-        ocpp_config["Internal"]["CentralSystemURI"] = f"{central_system_host}:{central_system_port}/{charge_point_id}"
-        return ocpp_config
+        return OCPPConfigAdjustmentVisitorWrapper(adjust_ocpp_configuration)
 
 
-class LibOCPP201ConfigurationHelper(LibOCPPConfigurationHelperBase):
-
+class _DefaultOCPP201ConfigurationVisitor(OCPPConfigAdjustmentVisitor):
```

Comment:
Could we make the name here a bit more descriptive? Or maybe add a docstring to explain what this visitor actually does?

Reply:
done